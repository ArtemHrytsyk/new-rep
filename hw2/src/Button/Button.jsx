import React from 'react';
import { useState } from 'react';


const Button = (props) => {
    const [textChange, textChangeHandler] = useState(props.text)

    return (
        <button onClick={() => { textChangeHandler('Thank you :)') }}>

            <span>{textChange}</span>
        </button>
    )
}

export default Button;