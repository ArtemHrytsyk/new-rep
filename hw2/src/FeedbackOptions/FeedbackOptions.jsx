import React from "react";

function FeedbackOptions({ goodFeedback, neutralFeedback, badFeedback }) {
    return (
        <React.Fragment>
            <button onClick={goodFeedback} className="button-good">Good:</button>
            <button onClick={neutralFeedback} className="button-neutral">Neutral:</button>
            <button onClick={badFeedback} className="button-bad">Bad:</button>
        </React.Fragment>
    );
}

export default FeedbackOptions;