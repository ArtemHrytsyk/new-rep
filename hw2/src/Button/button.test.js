import react from 'react'
import Button from './Button'
import renderer from 'react-test-renderer'
import React from 'react';
import FeedbackOptions from '../FeedbackOptions';

import Enzyme, { shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
Enzyme.configure({ adapter: new Adapter() })
import { render, screen } from '@testing-library/react';

import '@testing-library/jest-dom'

import '@testing-library/jest-dom'

describe('the OK button tests', () => {
    it('renders correctly', () => {
        const wrapper = renderer.create(<Button text='OK' />);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('renders with props text: OK', () => {
        const wrapper = renderer.create(<Button text='OK' />);
        const testInstance = wrapper.root;
        expect(testInstance.findByProps({ text: 'OK' })).toBeTruthy();
    });
});

