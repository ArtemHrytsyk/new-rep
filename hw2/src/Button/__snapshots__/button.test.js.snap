// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the OK button tests renders correctly 1`] = `
<button
  onClick={[Function]}
>
  <span>
    OK
  </span>
</button>
`;
